src/point-cloud-layer.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  CompositeLayer,
+  CompositeLayerProps,
+  DefaultProps,
+  GetPickingInfoParams,
+  Layer,
+  LayersList,
+  assert,
+  Unit,
+  Material,
+} from "@deck.gl/core/typed";
+import { PointCloudLayer } from "@deck.gl/layers/typed";
+import type { PointCloudLayerProps } from "@deck.gl/layers/typed";
+import * as arrow from "apache-arrow";
+import * as ga from "@geoarrow/geoarrow-js";
+import {
+  assignAccessor,
+  extractAccessorsFromProps,
+  getGeometryVector,
+} from "./utils.js";
+// TODO which accessors are actually needed for a pointcloud layer
+import {
+  GeoArrowExtraPickingProps,
+  computeChunkOffsets,
+  getPickingInfo,
+} from "./picking.js";
+import { ColorAccessor, GeoArrowPickingInfo, NormalAccessor } from "./types.js";
+import { EXTENSION_NAME } from "./constants.js";
+import { validateAccessors } from "./validate.js";
+
+/* All properties supported by GeoArrowPointCloudLayer */
+export type GeoArrowPointCloudLayerProps = Omit<
+  PointCloudLayerProps<arrow.Table>, // TODO is this still an Arrow Table or is it another datatype? An arrow table is a vector/array like value so is it the same?
+  "omitted properties here"
```

Comment:
We want to exclude any upstream types that we fully override. For example, looking at the `getNormal` type, if we don't exclude it from the upstream type, we end up with a type of our type `&` the upstream type, which is the intersection of the two types. 

<img width="813" alt="image" src="https://github.com/geoarrow/deck.gl-layers/assets/15164633/914d03ab-408a-4b40-83f8-bf8b73dddb2f">

So here we want to exclude `data`, `getNormal`, `getColor` and `getPosition`.